### DISCUSSION TITLE
Vue runtime does not export 'withScopeId'

### DISCUSSION THREAD

**Discussion by bgschiller** [parcel-bundler/parcel#6020] [Q&A]
I have an app bundled with parcel v1 that seems to be experiencing the issue described at #5160: "PostCSS plugin autoprefixer requires PostCSS 8. Update PostCSS or downgrade this plugin."

> 🚨  /Users/brian/code/URBANopt-web/ui/node_modules/vue-prism-editor/dist/VuePrismEditor.css:undefined:undefined: PostCSS plugin autoprefixer requires PostCSS 8. Update PostCSS or downgrade this plugin.

In trying to resolve the problem, I followed the advice in that issue to pin the versions of autoprefixer, postcss-modules, and parcel-bundler. Following that, I was able to compile successfully, but the app was missing most of my styling. It seemed to be ignoring the styles that came from within Vue SFCs.

Following what seems to be the recommendation of most parcel folks, I tried to upgrade to parcel 2. This took much of the day, as it seems like the Vue transformer is much more strict about requiring closing tags in components and also doesn't allow `~`-relative imports. On my parcel v2 branch, I'm now getting hundreds of errors like the following:

```
🚨 Build failed.
@parcel/core: node_modules/vue/dist/vue.runtime.esm.js does not export 'withScopeId'
/Users/brian/code/URBANopt-web/ui/src/App.vue:2:30
  1 | <template>
> 2 |   <div class="vue-entry-point-app"><!-- This is a vue component -->
>   |                              ^^^^^^^^^^^^
  3 |     <router-view></router-view>
  4 |   </div>
@parcel/core: node_modules/vue/dist/vue.runtime.esm.js does not export 'pushScopeId'
/Users/brian/code/URBANopt-web/ui/src/App.vue:4:1
  3 |     <router-view></router-view>
> 4 |   </div>
>   | ^^^^^^^^^
  5 | </template>
  6 | 
@parcel/core: node_modules/vue/dist/vue.runtime.esm.js does not export 'popScopeId'
/Users/brian/code/URBANopt-web/ui/src/App.vue:6:1
  5 | </template>
> 6 | 
>   | ^
  7 | <script>
  8 | export default {
@parcel/core: node_modules/vue/dist/vue.runtime.esm.js does not export 'resolveComponent'
/Users/brian/code/URBANopt-web/ui/src/App.vue:9:34
   8 | export default {
>  9 |   name: "App",
>    |               ^
  10 |   computed: {
  11 |     selectedFeatureIds() {
```

I'm imagining these SFCs were compiled (leading to the `withScopeId` and other functions) but the error message is still attached to the original source. Maybe the fix is as simple as switching module formats?

I'm hoping for a solution that works for either my v1 or v2 parcel branches. Thanks!

**Comment by devongovett** [comment]
cc @101arrowz any idea?

**Reply by 101arrowz** [reply]
Looks like a mismatch between the compiler version and the core runtime version? Not sure. Could you send your `package.json` and check the versions of all your Vue components?

**Reply by bgschiller** [reply]
happy to, thanks for taking a look! These are the package.json and package-lock.json from the parcel-v2 branch.

https://clip.brianschiller.com/2A2ZpwD-2021-03-19-package.json
https://clip.brianschiller.com/FFwvjBw-2021-03-19-package-lock.json

I feel a need to explain the ridiculous list of dependencies. It's an old angular.js app that is being converted to Vue piecemeal.

**Reply by 101arrowz** [reply]
The problem is that you're using Vue 2, but the Vue transformer supports only Vue 3. Please update to Vue 3 and everything should work fine.

**Reply by bgschiller** [reply]
ah, that makes sense. Unfortunately, I'm nervous about upgrading Vue until the migration build is available. I'll try and work around this some other way,
